src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/RangeExtensions.cs

```diff
@@ -139,26 +138,6 @@ public static TextSpan AsTextSpan(this Range range, SourceText sourceText)
         }
 
         return new TextSpan(start, length);
-
-        static int GetAbsolutePosition(Position position, SourceText sourceText, [CallerArgumentExpression(nameof(position))] string? argName = null)
-        {
-            var line = position.Line;
-            var character = position.Character;
-            var lineCount = sourceText.Lines.Count;
-            if (line > lineCount ||
-                (line == lineCount && character > 0))
-            {
-                throw new ArgumentOutOfRangeException($"{argName} ({line},{character}) matches or exceeds SourceText boundary {lineCount}.");
-            }
-
-            // LSP spec allowed a Range to end one line past the end, and character 0. SourceText does not, so we adjust to the final char position
-            if (line == lineCount)
-            {
-                return sourceText.Length;
-            }
```

Comment:
We definitely want to keep this check. `sourceText.GetAbsolutePosition` does not have it.